week1/matrix_done.py

```diff
@@ -0,0 +1,91 @@
+class Matrix:
+    def __init__(self, matrix_list):
+        self.rows = matrix_list
+
+    def __str__(self):
+        return '\n'.join(' '.join(map(str, sl)) for sl in self.rows)
+
+    def __eq__(self, other):
+        return self.rows == other.rows
+
+    def __add__(self, other):
+        m = Matrix([])
+        n = len(self.rows)
+        rows = [[0]*n for x in range(n)]
+        for i in range(n):
+            for j in range(n):
+                rows[i][j] = self.rows[i][j] + other.rows[i][j]
+                m.rows = rows
+        return m
+
+    def __mul__(self, other):
+        ro = len(other.rows)
+        co = len(other.rows[0])
+        cs = len(self.rows[0])
+        rs = len(self.rows)
+
+        m = Matrix.zeros(ro, co)
+        if ro != cs:
+            raise ValueError("cannot multiply matrices with not matching columns and rows")
+        else:
+            for i in range(rs):
+                for j in range(co):
+                    summa = 0
+                    for k in range(cs):
+                        summa = summa+self.rows[i][k]*other.rows[k][j]
+                        m.rows[i][j] = summa
+        return m
+
+    @staticmethod
```

Comment:
`@classmethod` is a much better fit here.

Please read online about the difference between `staticmethod` vs `classmethod `. Methods `Zeroes` and `Identity` are semantically classmethods, so make them classmethods please.